include/SDL_video_capture.h

```diff
@@ -0,0 +1,301 @@
+ *  \sa SDL_GetVideoCaptureFormat
+ *  \sa SDL_GetVideoCaptureFrameSize
+ *
+ */
+typedef struct SDL_VideoCaptureSpec
+{
+    Uint32 format;          /**< Video data format */
+    Uint32 width;           /**< Width */
+    Uint32 height;          /**< Height */
+    Uint32 pitch;           /**< Number of byte per line */
+    Uint32 size;            /**< Frame size */
+} SDL_VideoCaptureSpec;
+
+/**
+ *  SDL Video Capture Status
+ */
+typedef enum
+{
+    SDL_VIDEO_CAPTURE_INIT = 0, /**< Init, spec hasn't been set */
+    SDL_VIDEO_CAPTURE_STOPPED,  /**< Stopped */
+    SDL_VIDEO_CAPTURE_PLAYING   /**< Playing */
+} SDL_VideoCaptureStatus;
+
+/**
+ *  SDL Video Capture Status
+ */
+typedef struct SDL_VideoCaptureFrame
+{
+    void *pixels;               /**< Pointer to memory data */
+    int fd;                     /**< DMABUF file descriptor if available (-1 if not) */
+    void *clientbuffer;         /**< Android clientbuffer if available (NULL if not) */
+    void *internal;             /**< Private field */
+} SDL_VideoCaptureFrame;
+
+/**
+ * Open a Video Capture device
+ *
+ * \param device_name device name to open. Can be NULL.
+ *
+ * \returns device id, or SDL_VIDEO_CAPTURE_INVALID on error
+ *
+ * \sa SDL_GetVideoCaptureDeviceName
+ * \sa SDL_GetNumVideoCaptureDevices
+ */
+extern DECLSPEC SDL_VideoCaptureDeviceID SDLCALL SDL_OpenVideoCapture(const char *device_name);
+
+/**
+ * Set specification
+ *
+ * \param id video capture opened device id
+ * \param desired desired video capture spec
+ * \param obtained obtained video capture spec
+ * \param allowed_changes allow changes or not
+ *
+ * \returns 0 on success, negative value on error
+ */
+extern DECLSPEC int SDLCALL SDL_SetVideoCaptureSpec(SDL_VideoCaptureDeviceID id,
+                                                  const SDL_VideoCaptureSpec *desired,
+                                                  SDL_VideoCaptureSpec *obtained,
+                                                  int allowed_changes);
+
+/**
+ * Open a Video Capture device and set specification
+ *
+ * \param device_name device name to open. Can be NULL.
+ * \param desired desired video capture spec
+ * \param obtained obtained video capture spec
+ * \param allowed_changes allow changes or not
+ *
+ * \returns device id, or SDL_VIDEO_CAPTURE_INVALID on error
+ */
+extern DECLSPEC SDL_VideoCaptureDeviceID SDLCALL SDL_OpenVideoCaptureWithSpec(
+                                                  const char *device_name,
+                                                  const SDL_VideoCaptureSpec *desired,
+                                                  SDL_VideoCaptureSpec *obtained,
+                                                  int allowed_changes);
+/**
+ * Get device name
+ *
+ * \param index device index between 0 and num -1
+ *
+ * \returns device name, shouldn't be freed
+ *
+ * \sa SDL_GetNumVideoCaptureDevices
+ */
+extern DECLSPEC const char * SDLCALL SDL_GetVideoCaptureDeviceName(int index);
```

Comment:
I love to see it! Seems like lots of video capture APIs don't provide a way to get the device name. On Linux this is probably just /dev/video, but on Windows or Mac this could be nice and descriptive.

Reply:
yes..
on ios this is very verbose. on linux: /dev/video. on android just number ...
maybe this can be improved

Reply:
I don't have any ideas to make this better, I just wanted to give it some appreciation 😄

Maybe the header should specify UTF8 encoded string, which I assume is what you're using.